### PR TITLE
Add Why Dusty? section to docs

### PR DESCRIPTION
**Pull request by thieman**
@jsingle @paetling

### CODE REVIEW

File: docs/why-dusty.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Why Dusty?
+
+We built Dusty to be a great way to run a development environment, and we've tried to learn from
```
Comment (paetling) [inline docs/why-dusty.md]: run on.   environment. We've

File: docs/why-dusty.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+* **Isolation**: With Docker containers, each service you run is fully isolated. If you have three versions of
+Postgres running across production, Dusty can easily emulate that for you on your local environment. This would
+be very difficult with most Vagrant setups.
+* **Ease of Updating**: By default, Dusty makes sure it has the latest version of its specs whenever you
+run `dusty up`. If someone changes a service in your stack and pushes a change to the specs, Dusty will set that up
+for you automatically.
+* **Efficiency**: Dusty's mix and match capability allows you to run only the apps you need at any given
+time. Switching apps is very easy and supported out of the box. To do this in Vagrant would require heavy
+scripting through a configuration management tool.
```
Comment (paetling) [inline docs/why-dusty.md]: worth mentioning power use?

File: docs/why-dusty.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+We love [Docker Compose](https://docs.docker.com/compose/), and Dusty actually uses Compose to orchestrate
+container lifecycles. By itself, however, Compose has some shortcomings that we've tried to address with Dusty.
+
+* **OS X Support**: Using Docker on OS X introduces a virtual machine between your host OS and your containers,
+and navigating that extra layer is often painful. It takes a lot of effort to see a container's web service in a
+browser running on your Mac, or to get files between your Mac and the container. Dusty solves these problems (see
+following section on OS X Support).
+* **Simplified Specs**: Compose's specs model is fairly static. You write a large Composefile with all the containers for a
+specific bundle of apps. It's difficult to share service configuration across these large files. Dusty simplifies the
```
Comment (paetling) [inline docs/why-dusty.md]: configurations?

File: docs/why-dusty.md (hunk 2 above)
Comment (thieman) [inline docs/why-dusty.md]: wat

Comment (thieman) [inline docs/why-dusty.md]: oh nah i think that's covered

File: docs/why-dusty.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+specs model: each container is defined in exactly one place, then specs are stitched together at runtime based on the
+dependencies for the bundles you want to run.
```
Comment (paetling) [inline docs/why-dusty.md]: do we need the dependencies part after runtime.  I would favor ending on "then specs are stitched together at runtime"

File: docs/why-dusty.md (hunk 2 above)
Comment (thieman) [inline docs/why-dusty.md]: don't want to get into a pissing match where i'm defending tiny linux without a lot of data

Comment (paetling) [inline docs/why-dusty.md]: ok

File: docs/why-dusty.md (hunk 4 above)
Comment (thieman) [inline docs/why-dusty.md]: How about `Dusty simplifies the
specs model: each container is defined in exactly one place, then the specs you need are stitched together at runtime.`

**Comment by paetling** [comment]
Not worth contrasting with development directly on osx?

**Comment by thieman** [comment]
Was going to originally, but I couldn't really think of how to put it other than WHAT IS WRONG WITH YOU

File: docs/why-dusty.md (hunk 4 above)
Comment (paetling) [inline docs/why-dusty.md]: :+1:

**Pull request merged**